src/components/Avatar/Avatar.styles.ts

```diff
@@ -0,0 +1,48 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import { palette } from '../../util/colors';
+import { mergeStyleSets } from '@uifabric/styling';
+import { memoizeFunction } from '@uifabric/utilities';
+import { IPersonaCoinStyleProps, IPersonaCoinStyles } from 'office-ui-fabric-react/lib/PersonaCoin';
+import { AvatarSize, BorderType } from './Avatar';
+import { borders } from '../../util/styles/borders';
+
+export interface AvatarStyleProps {
+  borderType: BorderType;
+  size: AvatarSize;
+}
+
+export const getPersonaCoinGetStylesFunction = (
```

Comment:
1. should we memoize **all** getStyles functions? is it a problem to create new object instances on every render cycle?
2. can we use a better name here, maybe `getPersonalCoinSyltes`?

Reply:
I hear what you're saying. I'm wondering if you noticed that it returns a "getStyles" function, rather than the actual style object.  We pass it as a function to the fabric component, where it get's invoked with the `IPersonaCoinStyleProps`. 

I'm unsure if memoizing would be the right thing to do here, seems to me like it should maybe be a responsibility on the fabric side.

They allude to it here, but aren't yet doing it. 
https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/utilities/src/classNamesFunction.ts#L14

Reply:
I was just thinking about this a bit more, and I think it does make sense to memoize.  I'm going to work on this a bit more.